xde/src/xde.rs

```diff
@@ -287,15 +291,22 @@ fn get_xde_state() -> &'static XdeState {
 impl XdeState {
     fn new() -> Self {
         let ectx = Arc::new(ExecCtx { log: Box::new(opte::KernelLog {}) });
+        let mut dev_map = KRwLock::new(DevMap::default());
+        dev_map.init(KRwLockType::Driver);
```

Comment:
While we're here, why not make the KRwLockType an argument to `KRwLock::new()`, rather than requiring a separate `init()` call for all the consumers?

(Commented here, instead of the definition, because GH's PR tool sucks)

Reply:
I think the blocker here was having `KRwLock::new` be a `const fn` to make the old global static work at all. I think we should take the same tack as `KMutex` (i.e., not expose `type` at all for now) since the DDI only permits `RW_DRIVER`.